packages/core/core/src/services/document-service/events.ts

```diff
@@ -0,0 +1,73 @@
+import { curry } from 'lodash/fp';
+
+import { UID, Schema, Utils, Modules } from '@strapi/types';
+import { sanitize } from '@strapi/utils';
+
+import { getDeepPopulate } from './utils/populate';
+
+const ALLOWED_EVENTS = {
+  ENTRY_CREATE: 'entry.create',
+  ENTRY_UPDATE: 'entry.update',
+  ENTRY_DELETE: 'entry.delete',
+  ENTRY_PUBLISH: 'entry.publish',
+  ENTRY_UNPUBLISH: 'entry.unpublish',
+  ENTRY_DRAFT_DISCARD: 'entry.draft-discard',
+};
+
+type EventName = Utils.Object.Values<typeof ALLOWED_EVENTS>;
+
+const sanitizeEntry = async (
+  model: Schema.ContentType<any> | Schema.Component<any>,
+  entry: Modules.Documents.AnyDocument
+) => {
+  return sanitize.sanitizers.defaultSanitizeOutput(
+    {
+      schema: model,
+      getModel(uid) {
+        return strapi.getModel(uid as UID.Schema);
+      },
+    },
+    entry
+  );
+};
+
+/**
+ * Triggers an event.
+ *
+ * It will populate the entry if it is not a delete event.
+ * So the event payload will contain the full entry.
+ */
+const emitEntryEvent = async (
+  uid: UID.Schema,
+  eventName: EventName,
+  entry: Modules.Documents.AnyDocument
+) => {
+  const populate = getDeepPopulate(uid, {});
+  const model = strapi.getModel(uid);
+
+  const emitEvent = async () => {
```

Comment:
if you create a manager you can separate this function and the emitEntry one to avoid redefining it on each event emit

Reply:
created a simple event manager , wdyt? 
Now it's not using the strapi global, but it's only to be used on the document service repository, let me know if you see id adequate :)